### ISSUE TITLE
feat: Unified WasmEdge tools

### ISSUE THREAD

**Issue by hydai** [WasmEdge/WasmEdge#2226]
## Motivation

WasmEdge provides two tools in the release assets: `wasmedgec` and `wasmedge`. However, several tools are planned to be released, including `wasm-signature-related tools (wasmedge-sign and wasmedge-verify, a draft idea, not finalized)`, `wasmedge-test (which allows users to execute wast, wasm test script files)`, and more.

However, providing multiple tools will make it too complicated to use. That's why we want a simple entry point, `wasmedge`. As its subcommands, all the tools above should be collected into this new tool.

```
Before:
wasmedge
wasmedgec
wasmedge-sign
wasmedge-verify
wasmedge-test
After:
wasmedge, wasmedge run
wasmedge compile
wasmedge sign
wasmedge verify
wasmedge test
```

## Details

This is an LFX mentorship issue. Since it's a time-limited period to implement this feature, we will only handle the `wasmedge run` and `wasmedge compile` in the first stage. If the progress is good, we can move forward with various features.

#### Recommended Skills

1. Familiar with C++ programming language.
2. (Optional) WebAssembly Knowledge.

#### Application Process
Candidates who want to apply for this issue, please go to the LFX mentorship pages to apply. There are several requirements that the candidates must submit:
1. Apply this on the LFX mentorship website; the link is **[here](https://mentorship.lfx.linuxfoundation.org/project/2ebb39fd-3497-44f3-90d7-e95b444b2bc8)**.
4. Finish the pretest; the pretest is [here](https://github.com/WasmEdge/WasmEdge/discussions/2227).

### Implementation details

To implement this feature, please check the current usage of the `wasmedgec` tool. The new integrated tool should fully support all existing commands and features. For example, `wasmedgec` will change to `wasmedge compile` directly.

You may need the following information to finish this, including how wasmedge handles the command line (PO component), how wasmedge converts the cmd options into internal configurations, how wasmedge interacts with its private components, and how wasmedge emits error/warning messages if there is anything wrong.

**Comment by VaibhavMalik4187** [comment]
Hi @hydai, I'm interested in working on this project. I've already attempted the pretest. I'll apply for the mentorship once the applications are open, is there anything else that needs to be taken care of?
Thanks.

**Comment by hydai** [comment]
> I'll apply for the mentorship once the applications are open, is there anything else that needs to be taken care of?
> 

Please make sure you submit the applications on the LFX mentorship website once it's available. Thanks.

**Comment by VaibhavMalik4187** [comment]
> 
Sure, I'll keep that in mind

**Comment by hydai** [comment]
Fixed by #2363
